[Bug] fe oberver restart faild.

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

branch-1.2-lts

### What's Wrong?

3FE 3BE，1 follower，2 oberver
Restart Doris FE nodes in the following order: observe, observe, follower. After a few cycles, observe cannot be started.

### What You Expected?

Fix this bug.

### How to Reproduce?

Restart Doris FE nodes in the following order: observe, observe, follower. After a few cycles, observe cannot be started.

### Anything Else?

observe log

**Exception:**
2023-08-02 20:41:00,324 WARN (qtp976782276-82|82) [MetaService.put():161] failed to get remote file. url: http://10.0.0.54:8030/image?version=892
java.net.ConnectException: 拒绝连接 (Connection refused)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_272]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_272]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_272]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_272]
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1950) ~[?:1.8.0_272]
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1945) ~[?:1.8.0_272]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_272]
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1944) ~[?:1.8.0_272]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1514) ~[?:1.8.0_272]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1498) ~[?:1.8.0_272]
	at org.apache.doris.master.MetaHelper.getRemoteFile(MetaHelper.java:81) ~[doris-fe.jar:1.2-SNAPSHOT]
	at org.apache.doris.httpv2.meta.MetaService.put(MetaService.java:156) ~[doris-fe.jar:1.2-SNAPSHOT]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_272]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_272]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_272]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_272]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205) ~[spring-shade-5-1.0.2-SNAPSHOT.jar:1.0.2-SNAPSHOT]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150) ~[spring-shade-5-1.0.2-SNAPSHOT.jar:1.0.2-SNAPSHOT]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117) ~[spring-webmvc-5.3.22.jar:5.3.22]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:895) ~[spring-webmvc-5.3.22.jar:5.3.22]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808) ~[spring-webmvc-5.3.22.jar:5.3.22]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-5.3.22.jar:5.3.22]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1070) ~[spring-webmvc-5.3.22.jar:5.3.22]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:963) ~[spring-webmvc-5.3.22.jar:5.3.22]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006) ~[spring-webmvc-5.3.22.jar:5.3.22]
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898) ~[spring-webmvc-5.3.22.jar:5.3.22]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687) ~[javax.servlet-api-3.1.0.jar:3.1.0]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883) ~[spring-webmvc-5.3.22.jar:5.3.22]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[javax.servlet-api-3.1.0.jar:3.1.0]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1656) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:292) ~[websocket-server-9.4.48.v20220622.jar:9.4.48.v20220622]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1626) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.apache.doris.httpv2.interceptor.ServletTraceIterceptor.doFilter(ServletTraceIterceptor.java:55) ~[doris-fe.jar:1.2-SNAPSHOT]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:201) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1626) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-shade-5-1.0.2-SNAPSHOT.jar:1.0.2-SNAPSHOT]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117) ~[spring-shade-5-1.0.2-SNAPSHOT.jar:1.0.2-SNAPSHOT]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1626) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-shade-5-1.0.2-SNAPSHOT.jar:1.0.2-SNAPSHOT]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117) ~[spring-shade-5-1.0.2-SNAPSHOT.jar:1.0.2-SNAPSHOT]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1626) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-shade-5-1.0.2-SNAPSHOT.jar:1.0.2-SNAPSHOT]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117) ~[spring-shade-5-1.0.2-SNAPSHOT.jar:1.0.2-SNAPSHOT]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1626) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:552) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:600) ~[jetty-security-9.4.48.v20220622.jar:9.4.48.v20220622]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1440) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:505) ~[jetty-servlet-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1355) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.Server.handle(Server.java:516) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:487) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:732) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:479) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277) ~[jetty-server-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) ~[jetty-io-9.4.48.v20220622.jar:9.4.48.v20220622]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) ~[jetty-io-9.4.48.v20220622.jar:9.4.48.v20220622]
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) ~[jetty-io-9.4.48.v20220622.jar:9.4.48.v20220622]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338) ~[jetty-util-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315) ~[jetty-util-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173) ~[jetty-util-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131) ~[jetty-util-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409) ~[jetty-util-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883) ~[jetty-util-9.4.51.v20230217.jar:9.4.51.v20230217]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034) ~[jetty-util-9.4.51.v20230217.jar:9.4.51.v20230217]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_272]
Caused by: java.net.ConnectException: 拒绝连接 (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method) ~[?:1.8.0_272]
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[?:1.8.0_272]
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206) ~[?:1.8.0_272]
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188) ~[?:1.8.0_272]
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[?:1.8.0_272]
	at java.net.Socket.connect(Socket.java:607) ~[?:1.8.0_272]
	at sun.net.NetworkClient.doConnect(NetworkClient.java:175) ~[?:1.8.0_272]
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:463) ~[?:1.8.0_272]
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:558) ~[?:1.8.0_272]
	at sun.net.www.http.HttpClient.<init>(HttpClient.java:242) ~[?:1.8.0_272]
	at sun.net.www.http.HttpClient.New(HttpClient.java:339) ~[?:1.8.0_272]
	at sun.net.www.http.HttpClient.New(HttpClient.java:357) ~[?:1.8.0_272]
	at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1226) ~[?:1.8.0_272]
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1162) ~[?:1.8.0_272]
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1056) ~[?:1.8.0_272]
	at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:990) ~[?:1.8.0_272]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1570) ~[?:1.8.0_272]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1498) ~[?:1.8.0_272]
	at sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:3061) ~[?:1.8.0_272]
	at org.apache.doris.master.MetaHelper.getRemoteFile(MetaHelper.java:76) ~[doris-fe.jar:1.2-SNAPSHOT]
	... 69 more



try to restart
2023-08-02 20:59:57,806 INFO (main|1) [PaloFe.start():124] Palo FE starting...
2023-08-02 20:59:57,813 INFO (main|1) [FrontendOptions.analyzePriorityCidrs():107] configured prior_cidrs value: 10.0.0.0/16
2023-08-02 20:59:57,830 INFO (main|1) [FrontendOptions.init():87] local address: /10.0.0.76.
2023-08-02 20:59:57,969 INFO (main|1) [ConsistencyChecker.initWorkTime():106] consistency checker will work from 23:00 to 23:00
2023-08-02 20:59:58,785 INFO (main|1) [PrivTable.addEntry():83] add priv entry: global priv. host: %, user: root, priv: , set by resolver: false, domain user ident: null
2023-08-02 20:59:58,786 INFO (main|1) [PaloAuth.setPasswordInternal():1250] finished to set password for 'root'@'%'. is replay: true
2023-08-02 20:59:58,787 INFO (main|1) [PaloAuth.grantInternal():936] finished to grant privilege. is replay: true
2023-08-02 20:59:58,787 INFO (main|1) [PaloAuth.grantInternal():968] finished to grant resource privilege. is replay: true
2023-08-02 20:59:58,788 INFO (main|1) [PaloAuth.createUserInternal():772] finished to create user: 'root'@'%', is replay: true
2023-08-02 20:59:58,788 INFO (main|1) [PrivTable.addEntry():83] add priv entry: global priv. host: %, user: admin, priv: , set by resolver: false, domain user ident: null
2023-08-02 20:59:58,788 INFO (main|1) [PaloAuth.setPasswordInternal():1250] finished to set password for 'admin'@'%'. is replay: true
2023-08-02 20:59:58,788 INFO (main|1) [PaloAuth.grantInternal():936] finished to grant privilege. is replay: true
2023-08-02 20:59:58,788 INFO (main|1) [PaloAuth.grantInternal():968] finished to grant resource privilege. is replay: true
2023-08-02 20:59:58,789 INFO (main|1) [PaloAuth.createUserInternal():772] finished to create user: 'admin'@'%', is replay: true
2023-08-02 20:59:58,864 INFO (main|1) [Env.getHelperNodes():1228] get helper nodes: [10.0.0.76:9010]
2023-08-02 20:59:58,871 INFO (main|1) [Env.getClusterIdAndRole():1103] finished to get cluster id: 942552399, role: OBSERVER and node name: 10.0.0.76_9010_1690978062164
2023-08-02 20:59:58,878 INFO (main|1) [Env.loadImage():1668] image does not exist: /data/cdw/doris/fe/meta/image/image.0
2023-08-02 20:59:59,582 INFO (UNKNOWN 10.0.0.76_9010_1690978062164(-1)|1) [BDBEnvironment.setup():162] add helper[10.0.0.76:9010] as ReplicationGroupAdmin
2023-08-02 20:59:59,585 WARN (UNKNOWN 10.0.0.76_9010_1690978062164(-1)|1) [Env.notifyNewFETypeTransfer():2377] notify new FE type transfer: UNKNOWN
2023-08-02 20:59:59,621 WARN (RepNode 10.0.0.76_9010_1690978062164(-1)|55) [Env.notifyNewFETypeTransfer():2377] notify new FE type transfer: OBSERVER
2023-08-02 20:59:59,645 INFO (stateListener|67) [Env$4.runOneCycle():2400] begin to transfer FE type from INIT to UNKNOWN
2023-08-02 20:59:59,646 INFO (stateListener|67) [Env$4.runOneCycle():2487] finished to transfer FE type to UNKNOWN
2023-08-02 20:59:59,646 INFO (stateListener|67) [Env$4.runOneCycle():2400] begin to transfer FE type from UNKNOWN to OBSERVER
2023-08-02 20:59:59,649 INFO (replayer|70) [Env.replayJournal():2504] replayed journal id is 0, replay to journal id is 1233
2023-08-02 20:59:59,650 ERROR (replayer|70) [BDBJournalCursor.<init>():83] Can not find the key:1, fail to get journal cursor. will exit.
2023-08-02 21:00:18,105 INFO (main|1) [PaloFe.start():124] Palo FE starting...
2023-08-02 21:00:18,111 INFO (main|1) [FrontendOptions.analyzePriorityCidrs():107] configured prior_cidrs value: 10.0.0.0/16
2023-08-02 21:00:18,126 INFO (main|1) [FrontendOptions.init():87] local address: /10.0.0.76.
2023-08-02 21:00:18,267 INFO (main|1) [ConsistencyChecker.initWorkTime():106] consistency checker will work from 23:00 to 23:00
2023-08-02 21:00:19,001 INFO (main|1) [PrivTable.addEntry():83] add priv entry: global priv. host: %, user: root, priv: , set by resolver: false, domain user ident: null
2023-08-02 21:00:19,003 INFO (main|1) [PaloAuth.setPasswordInternal():1250] finished to set password for 'root'@'%'. is replay: true
2023-08-02 21:00:19,004 INFO (main|1) [PaloAuth.grantInternal():936] finished to grant privilege. is replay: true
2023-08-02 21:00:19,004 INFO (main|1) [PaloAuth.grantInternal():968] finished to grant resource privilege. is replay: true
2023-08-02 21:00:19,005 INFO (main|1) [PaloAuth.createUserInternal():772] finished to create user: 'root'@'%', is replay: true
2023-08-02 21:00:19,005 INFO (main|1) [PrivTable.addEntry():83] add priv entry: global priv. host: %, user: admin, priv: , set by resolver: false, domain user ident: null
2023-08-02 21:00:19,005 INFO (main|1) [PaloAuth.setPasswordInternal():1250] finished to set password for 'admin'@'%'. is replay: true
2023-08-02 21:00:19,005 INFO (main|1) [PaloAuth.grantInternal():936] finished to grant privilege. is replay: true
2023-08-02 21:00:19,005 INFO (main|1) [PaloAuth.grantInternal():968] finished to grant resource privilege. is replay: true
2023-08-02 21:00:19,005 INFO (main|1) [PaloAuth.createUserInternal():772] finished to create user: 'admin'@'%', is replay: true
2023-08-02 21:00:19,085 INFO (main|1) [Env.getHelperNodes():1228] get helper nodes: [10.0.0.76:9010]
2023-08-02 21:00:19,092 INFO (main|1) [Env.getClusterIdAndRole():1103] finished to get cluster id: 942552399, role: OBSERVER and node name: 10.0.0.76_9010_1690978062164
2023-08-02 21:00:19,098 INFO (main|1) [Env.loadImage():1668] image does not exist: /data/cdw/doris/fe/meta/image/image.0
2023-08-02 21:00:19,800 INFO (UNKNOWN 10.0.0.76_9010_1690978062164(-1)|1) [BDBEnvironment.setup():162] add helper[10.0.0.76:9010] as ReplicationGroupAdmin
2023-08-02 21:00:19,804 WARN (UNKNOWN 10.0.0.76_9010_1690978062164(-1)|1) [Env.notifyNewFETypeTransfer():2377] notify new FE type transfer: UNKNOWN
2023-08-02 21:00:19,834 WARN (RepNode 10.0.0.76_9010_1690978062164(-1)|55) [Env.notifyNewFETypeTransfer():2377] notify new FE type transfer: OBSERVER
2023-08-02 21:00:19,856 INFO (stateListener|67) [Env$4.runOneCycle():2400] begin to transfer FE type from INIT to UNKNOWN
2023-08-02 21:00:19,858 INFO (stateListener|67) [Env$4.runOneCycle():2487] finished to transfer FE type to UNKNOWN
2023-08-02 21:00:19,858 INFO (stateListener|67) [Env$4.runOneCycle():2400] begin to transfer FE type from UNKNOWN to OBSERVER
2023-08-02 21:00:19,873 INFO (replayer|70) [Env.replayJournal():2504] replayed journal id is 0, replay to journal id is 1254
2023-08-02 21:00:19,875 ERROR (replayer|70) [BDBJournalCursor.<init>():83] Can not find the key:1, fail to get journal cursor. will exit.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/fe/fe-core/src/main/java/org/apache/doris/master/Checkpoint.java
+++ b/fe/fe-core/src/main/java/org/apache/doris/master/Checkpoint.java
@@ -30,6 +30,7 @@ import org.apache.doris.persist.EditLog;
 import org.apache.doris.persist.MetaCleaner;
 import org.apache.doris.persist.Storage;
 import org.apache.doris.qe.VariableMgr;
+import org.apache.doris.service.FrontendOptions;
 import org.apache.doris.system.Frontend;
 
 import com.google.common.base.Strings;
@@ -51,6 +52,7 @@ public class Checkpoint extends MasterDaemon {
     private static final int PUT_TIMEOUT_SECOND = 3600;
     private static final int CONNECT_TIMEOUT_SECOND = 1;
     private static final int READ_TIMEOUT_SECOND = 1;
+    private static final int RETRY_TIME = 5;
 
     private Env env;
     private String imageDir;
@@ -183,26 +185,45 @@ public class Checkpoint extends MasterDaemon {
         int otherNodesCount = 0;
         if (!allFrontends.isEmpty()) {
             otherNodesCount = allFrontends.size() - 1; // skip master itself
-            for (Frontend fe : allFrontends) {
-                String host = fe.getHost();
-                if (host.equals(Env.getServingEnv().getMasterIp())) {
-                    // skip master itself
-                    continue;
+            int port = Config.http_port;
+            // make sure master FE http service is serving before pushing the image.
+            boolean isHttpServing = false;
+            for (int i = 0; i < RETRY_TIME && !isHttpServing; i++) {
+                isHttpServing = checkHttpServing(FrontendOptions.getLocalHostAddress(), port);
+                if (isHttpServing) {
+                    isHttpServing = true;
+                    break;
                 }
-                int port = Config.http_port;
-
-                String url = "http://" + host + ":" + port + "/put?version=" + replayedJournalId
-                        + "&port=" + port;
-                LOG.info("Put image:{}", url);
 
                 try {
-                    MetaHelper.getRemoteFile(url, PUT_TIMEOUT_SECOND * 1000, new NullOutputStream());
-                    successPushed++;
-                } catch (IOException e) {
-                    LOG.error("Exception when pushing image file. url = {}", url, e);
+                    Thread.sleep(3000);
+                } catch (InterruptedException e) {
+                    LOG.warn("InterruptedException", e);
                 }
             }
 
+            if (isHttpServing) {
+                for (Frontend fe : allFrontends) {
+                    String host = fe.getHost();
+                    if (host.equals(Env.getServingEnv().getMasterIp())) {
+                        // skip master itself
+                        continue;
+                    }
+
+                    String url = "http://" + host + ":" + port + "/put?version=" + replayedJournalId
+                            + "&port=" + port;
+                    LOG.info("Put image:{}", url);
+
+                    try {
+                        MetaHelper.getRemoteFile(url, PUT_TIMEOUT_SECOND * 1000, new NullOutputStream());
+                        successPushed++;
+                    } catch (IOException e) {
+                        LOG.error("Exception when pushing image file. url = {}", url, e);
+                    }
+                }
+            } else {
+                LOG.warn("push image error cause master FE http service is not serving");
+            }
             LOG.info("push image.{} to other nodes. totally {} nodes, push succeed {} nodes",
                     replayedJournalId, otherNodesCount, successPushed);
         }
@@ -346,5 +367,28 @@ public class Checkpoint extends MasterDaemon {
             long max = jvmStats.getMem().getHeapMax().getBytes();
             return used * 100 / max;
         }
+    }
+
+    /*
+     * Check if FE http service is serving
+     * Return true if http code is 200
+     */
+    private boolean checkHttpServing(String host, int port) {
+        URL idURL;
+        HttpURLConnection conn = null;
+        try {
+            idURL = new URL("http://" + host + ":" + port);
+            conn = (HttpURLConnection) idURL.openConnection();
+            conn.setConnectTimeout(CONNECT_TIMEOUT_SECOND * 1000);
+            conn.setReadTimeout(READ_TIMEOUT_SECOND * 1000);
+            return conn.getResponseCode() == 200;
+        } catch (Throwable e) {
+            LOG.warn("Exception when query FE http service, host: {}, port: {}, e: {}", host, port, e);
+        } finally {
+            if (conn != null) {
+                conn.disconnect();
+            }
+        }
+        return false;
     }
 }

